lib/internal/test_runner/test.js

```diff
@@ -300,8 +301,7 @@ class Test extends AsyncResource {
         ownAfterEachCount: 0,
       };
 
-      if ((testNamePatterns !== null && !this.matchesTestNamePatterns()) ||
-          (testOnlyFlag && !this.only)) {
+      if (!this.testMatchesCriteria() || (testOnlyFlag && !this.only)) {
```

Comment:
I would really prefer if this implementation doesn't change this much and `testNamePatterns` will just include the negated regexes of `testSkipPatterns`